algebra-core/src/main/scala/algebra/Priority.scala

```diff
@@ -0,0 +1,71 @@
+package algebra
+
+import scala.annotation.nowarn
+
+/**
+ * Priority is a type class for prioritized implicit search.
+ *
+ * This type class will attempt to provide an implicit instance of `P`
+ * (the preferred type). If that type is not available it will
+ * fallback to `F` (the fallback type). If neither type is available
+ * then a `Priority[P, F]` instance will not be available.
+ *
+ * This type can be useful for problems where multiple algorithms can
+ * be used, depending on the type classes available.
+ */
+@deprecated("No replacement", since = "2.7.0")
```

Comment:
why is this deprecated? It seems it is not deprecated in algebra?

The motivation is that some functions can be written that e.g. use either a Group or a Monoid. If you have a Group, it can be more efficient, or else you can fall back to Monoid (at a constant multiplicative cost).

Reply:
Note: we could imagine pushing this onto `Either[A, B]` in the scala standard library (prefer Right, fallback to Left). Similarly with `Option[A]` and tuples, but I think the standard library currently does not allow this.

Reply:
I was just replaying 7ff1809c065a020d269bc64e34a9a83f8e2c2417 which was suggested on the original PR since it's not used in algebra itself. My bad, now I see its intrinsic value. I'll bring it back.

Btw I did see your plans to submit it to Scala standard library in https://github.com/typelevel/algebra/issues/67!